tpm2tools/misc_test.go

```diff
@@ -0,0 +1,88 @@
+package tpm2tools
+
+import (
+	"bytes"
+	"crypto/sha1"
+	"crypto/sha256"
+	"fmt"
+	"hash"
+	"testing"
+
+	"github.com/google/go-tpm-tools/internal"
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+)
+
+var tests = []struct {
+	inAlg        tpm2.Algorithm
+	inExtensions [][]byte
+}{
+	{tpm2.AlgSHA1, nil},
+	{tpm2.AlgSHA1, [][]byte{bytes.Repeat([]byte{0x00}, sha1.Size)}},
+	{tpm2.AlgSHA1, [][]byte{bytes.Repeat([]byte{0x01}, sha1.Size)}},
+	{tpm2.AlgSHA1, [][]byte{bytes.Repeat([]byte{0x02}, sha1.Size)}},
+	{tpm2.AlgSHA256, nil},
+	{tpm2.AlgSHA256, [][]byte{bytes.Repeat([]byte{0x00}, sha256.Size)}},
+	{tpm2.AlgSHA256, [][]byte{bytes.Repeat([]byte{0x01}, sha256.Size)}},
+	{tpm2.AlgSHA256, [][]byte{bytes.Repeat([]byte{0x02}, sha256.Size)}},
+}
+
+func pcrExtend(alg tpm2.Algorithm, old, new []byte) ([]byte, error) {
+	var h hash.Hash
+	switch alg {
```

Comment:
This switch isn't needed, we have [`Algorithm.HashConstructor`](https://godoc.org/github.com/google/go-tpm/tpm2#Algorithm.HashConstructor)

Reply:
TBH, I might be confused, but I'd still need a switch in that case to convert the TPM alg type to the go alg type.

Reply:
Done.